vernac/lemmas.ml

```diff
@@ -420,18 +420,21 @@ let start_proof_with_initialization kind ctx decl recguard thms snl hook =
         | None -> Evd.empty_evar_universe_context
         | Some ctx -> ctx
         in
+        let evd = Evd.from_ctx ctx in
+        let uctx = Evd.check_univ_decl ~poly:(pi2 kind) evd decl in
+        let binders = Evd.universe_binders evd in
```

Comment:
These functions are also available in `UState`.

Reply:
This avoids creating a useless proxy evd indeed, will fix.